From 83881072f53904c0f3e1ad2f68658a557cb0bb34 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=D0=BD=D0=B0=D0=B1?= <nabijaczleweli@nabijaczleweli.xyz>
Date: Sat, 6 Nov 2021 18:27:00 +0100
Subject: [PATCH 4/4] browser.c: fix constant string warning
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Warning (GCC 11.2.0-10):
browser.c: In function ‘mutt_buffer_select_file’:
browser.c:1252:6: warning: offset ‘1’ outside bounds of constant string [-Warray-bounds]
 1252 | void mutt_buffer_select_file(struct Buffer *file, SelectFileFlags flags,
      |      ^~~~~~~~~~~~~~~~~~~~~~~
browser.c:1566:23: warning: offset ‘1’ outside bounds of constant string [-Warray-bounds]
 1566 |                   p = strrchr(mutt_buffer_string(&LastDir) + 1, '/');
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

This is valid because lastdirlen can only be >1
if there is at least one character in it,
so .data!=0, so we don't need to or want to fall back
to the empty string (which would segfault (preferable)
or just pick the next string in .rodata, which is fun)
---
 browser.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/browser.c b/browser.c
index 3d38520da..ef44954f2 100644
--- a/browser.c
+++ b/browser.c
@@ -1563,7 +1563,7 @@ void mutt_buffer_select_file(struct Buffer *file, SelectFileFlags flags,
               {
                 char *p = NULL;
                 if (lastdirlen > 1)
-                  p = strrchr(mutt_buffer_string(&LastDir) + 1, '/');
+                  p = strrchr(LastDir.data + 1, '/');
 
                 if (p)
                 {
-- 
2.33.1

